[vim] Improve typealias and associatedtype

tokorom opened the pull request:
**Update only utils/vim/syntax/swift.vim**

Improve the syntax highlight for `typealias` and `associatedtype`.

## Problematic patterns

![2018-10-02 8 45 25](https://user-images.githubusercontent.com/629993/46321750-1813bd00-c620-11e8-91dc-7529324e14db.png)

## Expects

![2018-10-02 8 47 20](https://user-images.githubusercontent.com/629993/46321754-1b0ead80-c620-11e8-80cc-59d24baecb0d.png)

tokorom committed:
[vim] Improve typealias and associatedtype

compnerd commented on utils/vim/syntax/swift.vim:
```diff
@@ -83,6 +82,7 @@ syn keyword swiftTypeDefinition skipwhite skipempty nextgroup=swiftTypeName
       \ protocol
       \ struct
       \ typealias
+      \ associatedtype
```
Please keep the list of keywords sorted.

compnerd commented:
Should we not highlight `associatedtype` as a keyword rather than a type identifier?  I guess you can argue either way.

tokorom committed:
[vim] Fix the list of keywords sorted

After the code review:
https://github.com/apple/swift/pull/19656#pullrequestreview-160565103

tokorom commented:
I read Declarations of Swift Language References and thought `typealias` and ` associatedtype` to be close. That's because both `typealias` and` associatedtype` has `typealias-name` immediately afterwards.

That is a reason I moved `associatedtype` to `swiftTypeDefinition`.

- [Type Alias Declaration](https://docs.swift.org/swift-book/ReferenceManual/Declarations.html#ID361)

> typealias-declaration → attributes access-level-modifier **typealias**  **typealias-name** generic-parameter-clause typealias-assignment

- [Protocol Associated Type Declaration](https://docs.swift.org/swift-book/ReferenceManual/Declarations.html#ID374)

> protocol-associated-type-declaration → attributes access-level-modifier **associatedtype** **typealias-name** type-inheritance-clause typealias-assignment generic-where-clause

![2018-10-02 16 08 25](https://user-images.githubusercontent.com/629993/46334634-c2f79b80-c65f-11e8-807c-011c0cffe8bf.png)
